Makefile

```diff
@@ -49,6 +49,7 @@ help: ## Display this help.
 .PHONY: manifests
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+	yq -i '.spec.versions[0].schema.openAPIV3Schema.properties.spec.properties.podTemplate.properties.spec.properties |= {}' config/crd/bases/etcd.aenix.io_etcdclusters.yaml
```

Comment:
I ran into the CRD being way too large as generated, so I borrowed the [approach from VictoriaMetrics](https://github.com/VictoriaMetrics/operator/blob/21ab04581f22d305d1e8265404fbfc5e4817ee25/Makefile#L71). Didn't manage to find a native way for kubebuilder to skip generation of a schema for nested fields of an embedded resource.

Reply:
Please, use `yq` as `$(YQ)` and add `yq` as a dependency to the makefile target.

Reply:
As an example:
https://github.com/aenix-io/etcd-operator/blob/4e3ddbe49ca00ae68bcab711627be60723019a0f/Makefile#L135-L141

Reply:
Also, could you please provide a link or smth. about the issue itself. It's hard to understand what is the real problem...

Reply:
> Also, could you please provide a link or smth. about the issue itself. It's hard to understand what is the real problem...

Here's the [problem](https://book.kubebuilder.io/faq.html?highlight=maxdesclen#the-error-too-long-must-have-at-most-262144-bytes-is-faced-when-i-run-make-install-to-apply-the-crd-manifests-how-to-solve-it-why-this-error-is-faced). It's caused by a very large CRD spec being written into the `kubernetes.io/last-applied-configuration` annotation. The reason why the CRD is so large is because the entire `corev1.PodSpec` resource is being embedded into it.

Reply:
Could we use server-side-apply to avoid this situation?

Reply:
> Could we use server-side-apply to avoid this situation?

There's a bunch more problems rising from this. For one, there's artifacthub's [size limits for helm charts (see linked discussion in Russian)](https://t.me/VictoriaMetrics_ru1/52767). What's worse, we don't have much control over how users install our operator and the CRDs. **We** can do SSA, sure, but there's no way to make sure that our users will do SSA.

Reply:
I am afraid to say it out loud. Should we redesign the spec then to avoid using podTemplate with k8s containers because it is too big?

Reply:
It sounds like a reasonable reason to abandon this idea.
I'm generally not opposed to adding fields to the spec when it is really necessary.

However, I would like to have a common logic for merging.

Reply:
With the post-processing, as presented in the diff to the Makefile, this becomes a non-issue. There's also no problem in taking the contents of the `PodTemplate` field and validating them according to the correct schema in a validating webhook.